src/ListTabulator/index.ts

```diff
@@ -366,61 +427,151 @@ export default class ListTabulator {
     if (event.isComposing) {
       return;
     }
+    if (currentItem === null) {
+      return;
+    }
+
+    const isEmpty = currentItem
+      ? this.renderer?.getItemContent(currentItem).trim().length === 0
+      : true;
+    const isFirstLevelItem = currentItem.parentNode === this.listWrapper;
+    const isLastItem = currentItem.nextElementSibling === null;
+    const hasSublist = currentItem.querySelector(`.${DefaultListCssClasses.itemChildren}`) !== null;
+
 
     /**
      * On Enter in the last empty item, get out of list
      */
-    const isEmpty = currentItem
-      ? this.list?.getItemContent(currentItem).trim().length === 0
-      : true;
-    const isFirstLevelItem = currentItem?.parentNode === this.listWrapper;
-    const isLastItem = currentItem?.nextElementSibling === null;
+    if (isFirstLevelItem && isEmpty) {
+      if (isLastItem && !hasSublist) {
+        this.getOutOfList();
+      }
+      /**
+       * If enter is pressed in the сenter of the list item we should split it
```

Comment:
right now split works only in empty item. I think, we should use split always when enter is pressed, even in non-empty item